### DISCUSSION TITLE
[#MVU] Updating model crashes on net8.0-windows10.0.19041 but works fine on net8.0-desktop

### DISCUSSION THREAD

**Discussion by kucint** [unoplatform/uno.extensions#2394] [Q&A] [answered]
I have created very simple app that presents a list of Items in an `ItemsRepeater`.
All works fine on `net8.0-desktop` but on net8.0-windows10.0.19041 the app constantly crashes when `MainPageModel.AddItem()` method is called.

Model:
```cs
internal record Item(string Text, Brush Brush);

internal partial record MainPageModel
{
  public IListState<Item> Items => ListState<Item>.Value(this, () =>
  [
      new Item("A", new SolidColorBrush(Colors.DeepSkyBlue)),
      new Item("B", new SolidColorBrush(Colors.DarkOrchid)),
      new Item("C", new SolidColorBrush(Colors.DarkOrange)),
  ]);
  
  public async ValueTask AddItem()
  {
      await Items.Update(items => items.Add(GetRandomItem()), CancellationToken.None).ConfigureAwait(true); // crashes
  }
}
```
View:
``` cs
new ItemsRepeater()
.ItemsSource(() => vm.Items)
.Layout(new StackLayout())
.ItemTemplate<Item>(item => GetItemView(item)),
 new Button()
 .Content("Add item")
 .Command(() => vm.AddItem)


private static TextBlock GetItemView(Item item) => new TextBlock()
    .Text(() => item.Text)
    .Foreground(() => item.Brush);
```
Note that the App crashes differently cross many executions which suggests some concurrency issue.
Note that another concurrency issue is as well reported in #2349. Maybe both are related?

_09:12:57:390	Exception thrown: 'System.Runtime.InteropServices.COMException' in System.Private.CoreLib.dll
09:12:57:390	Exception thrown: 'System.Runtime.InteropServices.COMException' in System.Private.CoreLib.dll
09:13:00:252	Exception thrown: 'System.Runtime.InteropServices.COMException' in WinRT.Runtime.dll
09:13:00:252	An exception of type 'System.Runtime.InteropServices.COMException' occurred in WinRT.Runtime.dll but was not handled in user code_

repro: [IListState.Update.crashes.zip](https://github.com/user-attachments/files/15983964/IListState.Update.crashes.zip)

**Comment by dansiegel** [comment]
cc: @dr1rrb

**Reply by morning4coffe-dev** [reply]
@kucint, @dr1rrb is currently looking into this.

**Reply by kucint** [reply]
great!
this is a regression, if I remember well, it used to work couple of dev branches ago ... and then it stopped.

**Reply by jeromelaban** [reply, marked as answer]
Let's track in https://github.com/unoplatform/uno.extensions/issues/2455.
